### PR TITLE
FIxed bugs in JupyterLabAppFactory

### PR DESCRIPTION
**Pull request by srinathnarayanan**
Fixed bug with logic to close the terminal tab once the shell is closed. If the shell hung, the earlier logic would not work. With these changes, we instead check for the opening of the shell instead of the closure, and then infer that the shell is closed if we see a new terminal line.

[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/1187)

### CODE REVIEW

File: src/Terminal/JupyterLabAppFactory.ts [hunk 1]
```diff
@@ -34,11 +46,12 @@ export class JupyterLabAppFactory {
     const session = await manager.startNew();
     session.messageReceived.connect(async (_, message: IMessage) => {
       const content = message.content && message.content[0]?.toString();
-      if (this.checkShellClosed && message.type == "stdout") {
+      console.log(content);
```
Comment (languy) [inline src/Terminal/JupyterLabAppFactory.ts:49]: Remove

**Review by languy — DISMISSED** [review]
Does it work as expected if shell takes a while to start or you need a bit of polling before deciding it's not started?

File: src/Terminal/JupyterLabAppFactory.ts (hunk 1 above)
Comment (srinathnarayanan) [reply src/Terminal/JupyterLabAppFactory.ts:49]: so the new terminal line (cosmosuser@...) is not seen once the shell is launched. If is only seen next after it shuts down or crashes in between.

**Comment by srinathnarayanan** [comment]
> Does it work as expected if shell takes a while to start or you need a bit of polling before deciding it's not started?

so the new terminal line (cosmosuser@...) is not seen once the shell is launched. It is only seen next after it shuts down or crashes in between.

**Review by languy — APPROVED** [review]

**Review by balalakshmin — APPROVED** [review]

**Pull request merged**